atox/src/main/kotlin/ui/call/CallFragment.kt

```diff
@@ -84,6 +85,15 @@ class CallFragment : BaseFragment<FragmentCallBinding>(FragmentCallBinding::infl
         updateSpeakerphoneIcon()
         speakerphone.setOnClickListener {
             vm.speakerphoneOn = !vm.speakerphoneOn
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                if (vm.speakerphoneOn) {
```

Comment:
I think this should go in the setter of `speakerphoneOn` in the viewmodel. The less logic in the GUI code, the better. Makes e.g. testing easier by allowing us to fake/mock viewmodels and things like that, as well as allowing us to test this behaviour without running a full GUI test.

Reply:
I don't feel that it belongs to the setter. But I agree that it should be moved to the viewmodel. So how about a dedicated method like `updateScreenOffWakelock()`?
Or one method that combines both the flipping between true and false in the `speakerphoneOn` and the wakelock handling like `toggleSpeakerphone()`? That way only this method and `updateSpeakerphoneIcon()` will be called in the onclick listener. It's like you wanted but just tells the reader that it's a function so it involves more logic than just toggling it.

Reply:
A toggleSpeakerphone-method is fine too! My main point was that it shouldn't live in the fragment. :P